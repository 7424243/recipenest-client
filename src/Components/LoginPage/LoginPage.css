input {
    margin-bottom: 10px;
}

input, label {
    display: block;
    width: 100%;
}

label {
    margin-bottom: 5px;
}


/* Classes */

.buttons {
    margin-top: 40px;
    text-align: center;
}

.login-form {
    margin: 0 auto;
    padding: 0 0 40px;
    max-width: 80%;
    text-align: left;
}


/* @media */

@media screen and (min-width: 540px) {
    .login-form {
        padding: 0 0 40px;
        max-width: 450px;
    }
}