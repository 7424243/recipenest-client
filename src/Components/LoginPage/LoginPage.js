import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../../services/token-service'
import config from '../../config'
import RecipenestContext from '../../RecipenestContext'
import PropTypes from 'prop-types'
import './LoginPage.css'

class LoginPage extends Component {

    //set default state
    state = {
        user_name: '',
        password: '',
        error: null
    }

    //allow access to context
    static contextType = RecipenestContext

    //Login with JWT authorization
    handleSubmitJwtAuth = e => {
        e.preventDefault()
        fetch(`${config.API_ENDPOINT}/auth/login`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
            },
            body: JSON.stringify(this.state)
        })
            .then(res => {
                if(!res.ok) {
                    return res.json().then(e => Promise.reject(e))
                }
                return res.json()
            })
            .then(res => {
                TokenService.saveAuthToken(res.authToken)
                this.context.onLoginSuccess()
                this.props.history.push('/my-recipes')
                
            })
            .catch(err => {
                this.setState({error: err.error})
                console.error({err})
            })
    }

    //update state's values for username and password
    handleUsername = e => {
        this.setState({user_name: e.target.value})
    }
    handlePassword = e => {
        this.setState({password: e.target.value})
    }


    render() {
        return (
            <>
                <h3>Login!</h3>
                <p className='demo-creds'>Demo Username: demo</p>
                <p className='demo-creds'>Demo Password: Password0!</p>
                <form 
                    className='login-form'
                    onSubmit={this.handleSubmitJwtAuth}
                >
                    <section className='username'>
                        <label htmlFor='username'>Username: </label>
                        <input 
                            type='text' 
                            placeholder='username' 
                            autoComplete='on'
                            required
                            onChange={this.handleUsername}
                        />
                    </section>
                    <section className='password'>
                        <label htmlFor='password'>Password: </label>
                        <input 
                            type='password' 
                            placeholder='password' 
                            autoComplete='on'
                            required
                            onChange={this.handlePassword}
                        />
                    </section>
                    <section className='buttons'>
                        <button type='submit' className='cancel-button'>Login</button>
                        <Link to='/signup'><button className='sign-up-button'>Sign Up</button></Link>
                        <Link to='/'><button className='save-button'>Cancel</button></Link>
                    </section>
                    {this.state.error && <p className='login-error'>{this.state.error}. Please try logging in again, or Sign Up for an account.</p>}
                </form>
            </>
        )
    }
}

export default LoginPage

LoginPage.propTypes = {
    history: PropTypes.shape({
        push: PropTypes.func
    }).isRequired
}