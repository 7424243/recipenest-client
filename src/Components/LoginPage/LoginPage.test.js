import React from 'react'
import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import LoginPage from './LoginPage';


describe('LoginPage Component', () => {

  const props = {
    history: {
        push: jest.fn()
    }
  }
  
  //smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><LoginPage {...props}/></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  
  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer 
      .create(<BrowserRouter><LoginPage {...props}/></BrowserRouter>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
  
})
