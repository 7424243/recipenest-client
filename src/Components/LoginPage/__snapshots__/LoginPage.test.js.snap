// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage Component renders the UI as expected 1`] = `
Array [
  <h3>
    Login!
  </h3>,
  <form
    className="login-form"
  >
    <section
      className="username"
    >
      <label
        htmlFor="username"
      >
        Username: 
      </label>
      <input
        placeholder="username"
        type="text"
      />
    </section>
    <section
      className="password"
    >
      <label
        htmlFor="password"
      >
        Password: 
      </label>
      <input
        placeholder="password"
        type="text"
      />
    </section>
    <section
      className="buttons"
    >
      <a
        href="/recipes"
        onClick={[Function]}
      >
        <button>
          Cancel
        </button>
      </a>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button>
          Sign Up
        </button>
      </a>
      <button>
        Submit
      </button>
    </section>
  </form>,
]
`;
